pkg/metrics/types/process_sample.go

```diff
@@ -43,3 +45,20 @@ type ProcessSample struct {
 	LastIOCounters  *process.IOCountersStat `json:"-"`
 	ContainerLabels map[string]string       `json:"-"`
 }
+
+// FlatProcessSample stores the process sampling information as a map
+type FlatProcessSample map[string]interface{}
+
+var _ sample.Event = &FlatProcessSample{} // FlatProcessSample implements sample.Event
+
+func (f *FlatProcessSample) Type(eventType string) {
```

Comment:
a Golang map is pointer to the underlying map struct, even being passed by value, what gets copied is the pointer, therefore method receivers don't need the pointer reference

Reply:
oh I see this struct was just moved

Reply:
This was a lift and shift of the original code. I didn't look to carefully at what it was doing when I moved it from [`pkg/metrics/process/sampler_linux.go`](https://github.com/newrelic/infrastructure-agent/blob/1.13.0/pkg/metrics/process/sampler_linux.go#L152) as was just trying to fix a circular reference when I moved it so didn't notice what it was doing 😅